src/framework/vst/internal/vstplugin.cpp

```diff
@@ -172,7 +172,7 @@ void VstPlugin::stateBufferFromString(VstMemoryStream& buffer, char* strData, co
 
     static Steinberg::int32 numBytesRead = 0;
 
-    buffer.write(strData, strSize, &numBytesRead);
+    buffer.write(strData, static_cast<Steinberg::int32>(strSize), &numBytesRead);
     buffer.seek(0, static_cast<size_t>(Steinberg::IBStream::kIBSeekSet), nullptr);

```

Comment:
While at it, actually the size_t cast in this line makes me wonder too... looks like it should not be there. 

And, one more thing: since `numBytesRead` is never read, it looks like it can be removed, and according to the documentation you can just pass nullptr instead of `&numBytesRead`.

Reply:
Hm, I don't immediately see what that compiler warning should be then... do you still get a warning nowadays when removing that size_t cast?

And `write()` does not read anything from the `numBytesRead` parameter (I've checked both the docs and the implementation in the VST API); it's really an "output parameter".

Reply:
Let's try

Reply:
Looks good!